sklearn/utils/extmath.py

```diff
@@ -530,21 +530,26 @@ def randomized_svd(
     )
 
     # project M to the (k + p) dimensional space using the basis vectors
-    B = Q.T @ M
+    B = Q.T @ M  # shape (k + p, M.shape[1])
 
-    # compute the SVD on the thin matrix: (k + p) wide
+    # Typically M.shape[1] > k + p, therefore we
+    # compute the SVD of the tall and thin matrix B.T with shape (M.shape[1], k + p)
+    # instead of less efficient SVD of (wide) matrix B.
```

Comment:
For reference, I ran the following quick experiment to check the empirical validity of the above comment:

```python
>>> import numpy as np
>>> a = np.random.normal(size=(10_000, 10))  # tall and thin
>>> %timeit np.linalg.svd(a, full_matrices=False)
1.67 ms ± 1.87 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
>>> %timeit np.linalg.svd(a.T, full_matrices=False)
3.01 ms ± 7.39 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)
>>> a.flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : False
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
```

Again, with a transposed memory layout:

```python
>>> b = np.random.normal(size=(10, 10_000))
>>> %timeit np.linalg.svd(b.T, full_matrices=False)  # tall and thin
1.61 ms ± 2.3 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
>>> %timeit np.linalg.svd(b, full_matrices=False)
2.99 ms ± 7.33 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)
>>> b.flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : False
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
```

Same conclusion: tall and thin is always the fastest, whatever the memory layout.

Reply:
I also tried with `scipy.linalg.svd(a, full_matrices=False, lapack_driver="gesvd")` and `scipy.linalg.svd(a, full_matrices=False, lapack_driver="gesdd")` instead of NumPy and got similar results.